spring-integration-core/src/test/java/org/springframework/integration/transformer/ToCloudEventTransformerTests.java

```diff
@@ -0,0 +1,114 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.transformer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.json.JsonPathUtils;
+import org.springframework.integration.test.util.TestUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.GenericMessage;
+
+import io.cloudevents.CloudEvent;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 5.3
+ */
+public class ToCloudEventTransformerTests {
+
+	private static final ConfigurableApplicationContext APPLICATION_CONTEXT = TestUtils.createTestApplicationContext();
+
+	private static final URI SOURCE = URI.create("https://spring.io/projects/spring-integration");
+
+	@AfterAll
+	static void teardown() {
+		APPLICATION_CONTEXT.close();
+	}
+
+	@Test
+	void testDefaultTransformer() {
+		ToCloudEventTransformer transformer = new ToCloudEventTransformer(SOURCE);
+		Message<?> result = transformer.transform(new GenericMessage<>("test"));
+		assertThat(result.getHeaders()).containsOnlyKeys(MessageHeaders.ID, MessageHeaders.TIMESTAMP);
+		assertThat(result.getPayload())
+				.asInstanceOf(InstanceOfAssertFactories.type(CloudEvent.class))
+				.satisfies(event -> {
+							assertThat(event.getData().get()).isEqualTo("test");
+							assertThat(event.getAttributes().getSource()).isEqualTo(SOURCE);
+							assertThat(event.getAttributes().getSpecversion()).isEqualTo("1.0");
+							assertThat(event.getAttributes().getType()).isEqualTo(String.class.getName());
+							assertThat(event.getAttributes().getMediaType().isPresent()).isFalse();
+						}
+				);
+	}
+
+	@Test
+	void testBinary() {
+		ToCloudEventTransformer transformer =
+				new ToCloudEventTransformer(SOURCE, ToCloudEventTransformer.Result.BINARY);
+		transformer.setSubjectExpression(new LiteralExpression("some_subject"));
+		GenericMessage<String> message = new GenericMessage<>("test");
+		Message<?> result = transformer.transform(message);
+		assertThat(result.getHeaders())
+				.containsEntry("ce_type", String.class.getName())
```

Comment:
note these all need to be `ce-*` not `ce_` if this is for http.

Reply:
Well, one more time. What we do here is a general framework approach when you would like to build a `CloudEvent` and  carry it from component to component as our `Message<?>` content.
This is not about protocol binding per se.
Of course when we come with this content to the HTTP level we will remap these `ce_` to `ce-`.

I have designed this after this library general approach for target protocol header mapping, e.g. https://github.com/spring-projects/spring-integration/blob/master/spring-integration-http/src/main/java/org/springframework/integration/http/HttpHeaders.java#L26, https://github.com/spring-projects/spring-integration/blob/master/spring-integration-redis/src/main/java/org/springframework/integration/redis/support/RedisHeaders.java#L34 and so on.
Plus I have looked to CE Kafka protocol binder and this is that prefix: https://github.com/cloudevents/sdk-java/blob/master/kafka/src/main/java/io/cloudevents/v1/kafka/HeaderMapper.java#L25